pbj-core/pbj-compiler/src/main/java/com/hedera/pbj/compiler/impl/Common.java

```diff
@@ -501,6 +501,168 @@ private static String getRepeatedEqualsGeneration(String generatedCodeSoFar, Fie
 		return generatedCodeSoFar;
 	}
 
+
+	/**
+	 * Generate the compareTo method content for the provided fields
+	 * @param fields The fields of this object.
+	 * @param generatedCodeSoFar the generated code so far (non-empty in case of nested objects)
+	 * @return The generated code for compareTo method body
+	 */
+	public static String getFieldsCompareToStatements(final List<Field> fields, String generatedCodeSoFar) {
+		for(Field f : fields) {
+			if (f.parent() != null) {
+				final OneOfField oneOfField = f.parent();
+				generatedCodeSoFar += getFieldsCompareToStatements(oneOfField.fields(), generatedCodeSoFar);
+			}
+
+			if (f.optionalValueType()) {
+				generatedCodeSoFar += getPrimitiveWrapperCompareToGeneration(f);
+			}
+			else if (f.repeated()) {
+				// intentionally skipping repeated fields
```

Comment:
If we do not support repeated fields for comparison (which is entirely reasonable), then we should refuse to compile any message with repeated fields that requests to implement `Comparable`.  That makes it clear both for the message author (via a compile error) and for the user of the Java class that those objects should not be used in sorted structures.

Reply:
@anthony-swirldslabs 

>one can always sort all the repeated values in both the objects, and then compare the sorted arrays lexicographically.

Even though it's certainly possible, it seems impractical for performance reasons. 

> we could throw an exception

Yes, I will modify `LookupHelper#extractComparableFields` accordingly

Reply:
@jsync-swirlds 
> we should refuse to compile any message with repeated fields that requests to implement `Comparable`
If the request is explicit, I agree - it should throw an exception. By default, I think it's reasonable to omit it from `compareTo`.

Reply:
The problem with a "default" compareTo that short-cuts repeated fields like this is that there will soon be code that checks if Comparable is implemented; if it is implemented then that code must assume the implementation is _correct_.  This will not be correct by default as it will not impose a consistent total order (as required by the [Comparable interface](https://docs.oracle.com/en/java/javase/21/docs/api/java.base/java/lang/Comparable.html)) on records with repeated fields.

Simply not supporting repeated fields (and only writing a compareTo if Comparable is explicitly requested) helps ensure we don't provide a comparison method to consuming code that is likely to cause unexpected failures.

Reply:
I think we should only implement Comparable on messages that have been annotated with the right comment and in that case we throw an error if the object cannot be made comparable.

Reply:
@anthony-swirldslabs @jsync-swirlds @rbair23 

Fair enough. I made changes to the implementation.
The current state of things:
- `Bytes` field _is_ supported. These fields are compared byte-by-byte
- OneOf is now properly supported. All the kinds of `OneOf` must implement `Comparable`. There is a validation for that. If it's not the case, the build fails.
- The order of fields is `pbj.comparable` option is preserved. Now they are used in `compareTo` in the order they are listed.